File: 160-intersection-of-two-linked-lists/intersection-of-two-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        HashSet<ListNode> hset = new HashSet<>();
        ListNode currentA = headA;
        while(currentA!=null){
            hset.add(currentA);
            currentA = currentA.next;
        }
        ListNode currentB = headB;
        while(currentB!=null){
            if(hset.contains(currentB)){
                return currentB;
            }
            currentB=currentB.next;
        }
        return null;
        
    }
}

File: 160-intersection-of-two-linked-lists/intersection-of-two-linked-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */
var getIntersectionNode = function(headA, headB) {
    let set = new Set();
     let currentA = headA;
        let currentB = headB;


     while(currentB!=null){
               set.add(currentB);
               currentB= currentB.next;
            }
            while(currentA!=null){
                if(set.has(currentA)){
                
                    return currentA;

                   
                }
                currentA = currentA.next;
            }

            return null;
    
};